compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5525,9 +5525,33 @@ TR::Register *commonLoadEvaluator(TR::Node *node, TR::InstOpCode::Mnemonic op, i
     TR::MemoryReference *tempMR = TR::MemoryReference::createWithRootLoadOrStore(cg, node);
     tempMR->validateImmediateOffsetAlignment(node, size, cg);
 
-    generateTrg1MemInstruction(cg, op, node, targetReg, tempMR);
+    if (needSync && tempMR->getUnresolvedSnippet() == NULL && size <= 4) {
```

Comment:
Is there any reason for limiting the size to 4 bytes or smaller?  What about `lload` and `dload`?

Reply:
No that is a mistake I will change it to 8